### DISCUSSION TITLE
NextJs hydration blocks my page for a few moments

### DISCUSSION THREAD

**Discussion by ziyodbek0300** [vercel/next.js#52821] [Help]
### Summary

First image:
![Screenshot 2023-07-18 at 15 17 13](https://github.com/vercel/next.js/assets/57743123/f687f03a-3640-41cd-b607-e4a26a57589f)

Second image: 
![Screenshot 2023-07-18 at 15 17 35](https://github.com/vercel/next.js/assets/57743123/ac1f504b-aaf1-45cc-8821-d6cd41eeff12)

**Comment by icyJoseph** [comment]
Is this in pages directory? I think those timing labels are not used in app dir, but I could be wrong.

Regardless, this is the problem that App Dir set out to solve, and in general, a potential improvement driven all the way from the React team. Hydration blocks and must complete the entire three. 

> You have to hydrate everything before you can interact with anything
> There is a similar issue with hydration itself. Today, React hydrates the tree in a single pass. This means that once it’s started hydrating (which is essentially calling your component functions), React won’t stop until it’s finished doing this for the entire tree. As a consequence, you have to wait for all components to be hydrated before you can interact with any of them. - https://github.com/reactwg/react-18/discussions/37

**Reply by ziyodbek0300** [reply]
Overall I can't do anything to improve rendering?

**Reply by icyJoseph** [reply]
You can, but it depends on the application. 

Generally speaking, small `__NEXT_DATA__` (a script tag of type application/json), and shipping less JS, installing less packages, will do a lot already.

- https://pkg-size.dev/
- https://bundlephobia.com/

**Reply by ziyodbek0300** [reply]
I got you. Thanks
